subiquity/models/tests/test_subiquity.py

```diff
@@ -185,18 +187,31 @@ def test_storage_version(self):
         config = model.render()
         self.assertConfigHasVal(config, "storage.version", 1)
 
-    def test_write_netplan(self):
-        model = self.make_model()
-        config = model.render()
-        netplan_content = None
-        for fspec in config["write_files"].values():
-            if fspec["path"].startswith("etc/netplan"):
-                if netplan_content is not None:
-                    self.fail("writing two files to netplan?")
-                netplan_content = fspec["content"]
-        self.assertIsNot(netplan_content, None)
-        netplan = yaml.safe_load(netplan_content)
-        self.assertConfigHasVal(netplan, "network.version", 2)
+    @parameterized.expand(
+        [
+            [{}, True],
+            [{"NETPLAN_CONFIG_ROOT_READ_ONLY": False}, True],
+            [{"NETPLAN_CONFIG_ROOT_READ_ONLY": True}, False],
+        ]
+    )
+    def test_write_netplan(self, features, np_content_expected):
+        with unittest.mock.patch(
+            "subiquity.cloudinit.open",
+            mock.mock_open(read_data=json.dumps({"features": features})),
+        ):
+            model = self.make_model()
+            config = model.render()
+            netplan_content = None
+
+            for fspec in config["write_files"].values():
+                if fspec["path"].startswith("etc/netplan"):
+                    if netplan_content is not None:
+                        self.fail("writing two files to netplan?")
+                    netplan_content = fspec["content"]
+            self.assertEqual(np_content_expected, netplan_content is not None)
+            if np_content_expected:
+                netplan = yaml.safe_load(netplan_content)
+                self.assertConfigHasVal(netplan, "network.version", 2)
```

Comment:
should we have another case here that asserts netplan_content is None if np_content_expected is false?

Reply:
I think that's covered already, pushed a tweak that is a bit more readible.  If this isn't what you meant please clarify.

Reply:
thanks!